components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -1763,12 +1792,12 @@ public void deleteGroup(String groupId) throws NotFoundException, CharonExceptio
             throw new NotImplementedException(error);
         }
 
-        if (SCIMCommonUtils.isFilteringEnhancementsEnabled()) {
+        if (isFilterUsersAndGroupsOnlyFromPrimaryDomainEnabled()) {
```

Comment:
This is checking for two scenarios to prepend the primary domain. Shall we follow a similar approach to https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/165/files#r279851284